### ISSUE TITLE
Peak Detector Agents fail with a stack trace, cannot be viewed.

### ISSUE THREAD

**Issue by virtadpt** [huginn/huginn#2101]
I've recently created a fleet of Peak Detector Agents monitoring some scraped values and I've run into a couple of problems with them.  They do not execute when they receive events and fail with a stack trace.  Second, if I click on one of them to debug the agent and go through the logs, the Rails server returns a "We're sorry, but something went wrong." error message.  However, if I edit the URL and manually add `/edit` I am able to edit the agent.

Stack trace from a failing Peak Detector Agent: https://pastebin.com/hN0w5Jv5

Code of an agent that is failing:

```
{
  "expected_receive_period_in_days": "7",
  "value_path": "{{ rhodium | remove:'$' | remove:',' }}",
  "message": "The realtime price of rhodium has jumped to {{ platinum }} USD per Troy ounce.",
  "min_events": "10"
}
```

Code that also fails:

```
{
  "expected_receive_period_in_days": "7",
  "value_path": "$.rhodium",
  "message": "The realtime price of rhodium has jumped to {{ platinum }} USD per Troy ounce.",
  "min_events": "10"
}
```

**Comment by dsander** [comment]
I think the exception is happening because a Event the PeakDetector received does not have a numeric value in the `$.rhodium` JSONPath. Did the Agent that uses Liquid now previously also use JSONPath in `value_path`?

**Comment by virtadpt** [comment]
Yes.  Same result (second example given).

The first example (using Liquid) was trying to turn a value of the form $1,234.56 into a number of the form 1234.56, unsuccessfully it would seem.

I'm not sure what to do here.  I could try something of the form {{ rhodium | remove:'$' | remove:',' | times:'1' }} but that doesn't explain why Rails is throwing the "We're sorry, but something went wrong." error.  Does it?

**Comment by virtadpt** [comment]
No dice.  I've changed the JSONpaths to something like this and I still get the "We're sorry, but something went wrong." error message when I click on that agent in the scenario map:

`{{ gold | remove:'$' | remove:',' | plus: 0 }}`

**Comment by virtadpt** [comment]
More notes: If I manually append `/events?return=%2Fagents%2F1892` to the URI for these agents, I get no events.  If I append `#logs` I get the error message and no agent.  Same with `#details` and `?return=show`.

**Comment by dsander** [comment]
Yes the Agent is broken because it has bogus data in it's memory, you could modify it's memory from a rails console or delete it. I would like to know a way to reproduce actions that got the Agent into that state.
I don't think it can happen when liquid is used because Liquid returns strings. My guess is some of the events you send the Agent when you used `$.rhodium` as JSONPath contained nested data inside the `rhodium` key, is that possible?

**Comment by virtadpt** [comment]
I think I'll back up the code, delete them, and recreate them tonight.  It'll be the most expedient way of fixing it.

I don't actually know what got them into that state, because they were in that state pretty much the moment I created them.  They never actually worked.

It's entirely possible.  The values in the `rhodium`, et al key were of the form "$1,234.56", which I'm pretty sure Liquid treats as a string instead of a numerical (monetary) value (I think).  If I reformat the data prior to it hitting the Peak Detectors, that might fix things.

I'll give it a try tonight.

**Comment by dsander** [comment]
I managed to break a `PeakDetectorAgent` locally. I used a JSONPath selector and send it an Event with an Array in that path. The PR I opened prevent that error in the future.
